File: node/src/main/java/com/shishkin/service/FileService.java
package com.shishkin.service;

import com.shishkin.common.jpa.entity.DocumentEntity;
import com.shishkin.common.jpa.entity.PhotoEntity;
import com.shishkin.exception.UploadFileException;
import org.telegram.telegrambots.meta.api.objects.Message;

public interface FileService {
    DocumentEntity processDoc(Message message) throws UploadFileException;
    PhotoEntity processPhoto(Message message) throws UploadFileException;
}

File: rest-service/src/main/java/com/shishkin/service/FileService.kt
package com.shishkin.service

import com.shishkin.common.jpa.entity.BinaryContent
import com.shishkin.common.jpa.entity.DocumentEntity
import com.shishkin.common.jpa.entity.PhotoEntity
import org.springframework.core.io.FileSystemResource

interface FileService {
    fun getDocument(id: String): DocumentEntity?
    fun getPhoto(id: String): PhotoEntity?
    fun getFileSystemSource(binaryContent: BinaryContent): FileSystemResource?
}